File: Shared/Requests/TemperatureRequest.cs
namespace Shared.Requests
{
public class TemperatureRequest : BaseRequest
{
    public List<double> Temperatures { get; set; } = [];

    public TemperatureRequest()
    {
    }

    public TemperatureRequest(TemperatureRequest request) : base(request)
    {
        Temperatures = request.Temperatures;
    }
}
}

File: Web/src/models/requests/TemperatureRequest.ts
import { BaseRequest } from "./BaseRequest";

export class TemperatureRequest extends BaseRequest {
  public temperatures: number[] = [];

  public constructor(from: string, to: string, temperatures?: number[]) {
    super(from, to);

    if (temperatures) {
      this.temperatures = temperatures;
    }
  }

  public override Initialize(data: number[]) {
    if (!data) {
      throw new Error("The temperatures are null!");
    }

    this.temperatures = data;
  }
}
